eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -1,5 +1,6 @@
 parameters:
   ServiceDirectory: ''
+  CoverageGoal: 0.95
```

Comment:
As the coverage go tool gains complexity/scope, I think it would be good to follow a convention of an SDK directory config file to store this type of info. As it is in this PR, you can't run the coverage locally to see if it will pass upstream without knowing to manually look up the CoverageGoal in ci.yml. It gets even harder if you want to run coverage for the whole repo (e.g. for cross-cutting changes).

Reply:
My thinking in general for go is that we want to put as little context in the yaml as possible. In an ideal world the devops yaml would only function as a light orchestration wrapper and not contain any sdk-specific context.

Reply:
Is there an example of another repo that has a config file? Or to store this in a single config file and specifying the information per library?

Reply:
I would say it's an open question, perhaps just a config.json file for easy interoperability? I know js uses the convention of putting custom key/value config data into the `package.json` file (e.g. for [samples configuration](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/eventhub/event-hubs/package.json#L84-L92)). I don't think there are as many examples in other repos.

If you think it's weird/unnecessary, I wouldn't disagree as right now there would be only tool that uses it. As we add more tooling that needs to have per-sdk behavior it could be helpful to have a common place for this kind of stuff that is amenable to both local and pipeline scripts.

Reply:
@benbp I added a config.json file in the `eng` directory. Right now it just has a list of packages and their code coverage goal but I think this will be a good way to expand any needs for it in the future.

Reply:
I like it!

Reply:
If it's going to be in eng and contain all package info, it can probably be a dedicated file called `coverage.json` instead?

Reply:
I went  with config.json instead because it may contain more info in the future other than coverage information.